### PR TITLE
Gruber Calculator

### PR DESCRIPTION
**Pull request by cpgruber**
Comfort_level: 4
Completeness: 5

Couldn't figure out Bonus 2... I think it would involve Regex, which is crazy.

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// I changed the caclulator to accept 1 or 2 arguments.
+// If the user enters a single argument, the operation would be performed on the
+// running value. If the user enters two arguments the running value resets to the
+// result of that operation.
+
+var calculator = {};
+calculator.value = 0;
+calculator.add = function(num1,num2){
+  if(num2){
```
Comment (jshawl) [inline calculator.js:9]: yes! This is the interface I had in mind as wel..

File: calculator.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  }
+  return calculator.value;
+}
+calculator.exponent = function(num1,num2){
+  if(num2){
+    calculator.value = Math.pow(num1,num2);
+  }else{
+    calculator.value = Math.pow(calculator.value,num1);
+  }
+  return calculator.value;
```
Comment (jshawl) [inline calculator.js:46]: maybe a ternary here could help clean up your code:

``` js
calc.exp = function( num1, num2 ){
  return calculator.value = num2 ? Math.pow(num1,num2) : Math.pow(this.value,num1);
}
```

File: calculator.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    calculator.value%= num1;
+  }
+  return calculator.value;
+}
+calculator.clear = function(){
+  calculator.value = 0;
+  return calculator.value;
+}
+calculator.master = function(string){
+  console.log("this part is hard...")
```
Comment (jshawl) [inline calculator.js:61]: what if you start with `eval`?:

``` js
calculator = {
    master: function( expression ){
        return eval(expression)
    }
}

calculator.master("(2/3) * 100")
```

**Comment by jshawl** [comment]
:+1:

File: calculator.js (hunk 3 above)
Comment (cpgruber) [inline calculator.js:61]: I did not know about eval. I thought you'd have to parse out parens and math operators somehow, but yea, eval definitely seems to be the simplest way here.

**Pull request closed without merging**